server/common/storage/index.js

```diff
@@ -84,27 +108,7 @@ class Storage {
   }
 }
 
-function createStorage(options = {}) {
-  // Validate provider name.
-  const providerName = options.provider;
-  if (!options[providerName]) {
-    throw new Error('Provider should be defined in config');
-  }
-
-  // Validate provider config.
-  const config = options[providerName];
-  const provider = loadProvider(providerName);
-  provider.schema.validate(config, { stripUnknown: true }, err => {
-    if (err) throw new Error('Unsupported config structure');
-  });
-
-  // Create store & client instance.
-  const store = provider.createStore(config);
-  const { errors } = provider;
-  return new Storage(store, { errors });
-}
-
-module.exports = createStorage;
+module.exports = new Storage(storage);
```

Comment:
Storage service for repository content should be a separate module. One might use this adapter to create multiple services. Revert to previous state and implement content storage service as a separate module. Within it you can implement stuff you implemented as 'helpers' above.